CoursesManager.UI/ViewModels/Courses/CourseOverViewViewModel.cs

```diff
@@ -117,12 +131,86 @@ private void OnCheckboxChanged(CourseStudentPayment payment)
+                        new DialogResultType
+                        {
+                            DialogText = "Cursus heeft nog actieve registraties.",
+                            DialogTitle = "Error"
+                        });
+                }
+                else
+                {
+                    var result = await _dialogService.ShowDialogAsync<ConfirmationDialogViewModel, DialogResultType>(
+                        new DialogResultType
+                        {
+                            DialogTitle = "Bevestiging",
+                            DialogText = "Weet je zeker dat je deze cursus wilt verwijderen?"
+                        });
+
+                    if (result.Outcome == DialogOutcome.Success && result.Data is not null && result.Data.Result)
+                    {
+                        try
+                        {
+                            _courseRepository.SetInactive(CurrentCourse);
+
+                            _messageBroker.Publish(new CoursesChangedMessage());
+                            _navigationService.GoBackAndClearForward();
+                        }
+                        catch (Exception ex)
+                        {
+                            LogUtil.Error(ex.Message);
+                            await _dialogService.ShowDialogAsync<ErrorDialogViewModel, DialogResultType>(
+                                new DialogResultType
+                                {
+                                    DialogText = "Er is iets fout gegaan.",
+                                    DialogTitle = "Error"
+                                });
+                        }
+                    }
+                }
+            });
+        }
+
+        private async void ChangeCourse()
         {
+
+            await ExecuteWithOverlayAsync(async () =>
+            {
+                var dialogResult = await _dialogService.ShowDialogAsync<CourseDialogViewModel, Course>(CurrentCourse);
+
+                if (dialogResult.Outcome == DialogOutcome.Success)
+                {
+                    CurrentCourse = dialogResult.Data;
+                }
+            });
         }
 
-        private void DeleteCourse()
+        private async Task ExecuteWithOverlayAsync(Func<Task> action)
         {
+            _messageBroker.Publish(new OverlayActivationMessage(true));
+            try
+            {
+                await action();
+            }
+            finally
+            {
+                _messageBroker.Publish(new OverlayActivationMessage(false));
+            }
         }
+
+        private System.Windows.Visibility isImageVisible;
+
+        public System.Windows.Visibility IsImageVisible { get => isImageVisible; set => SetProperty(ref isImageVisible, value); }
+
+        private System.Windows.Visibility isPlaceholderVisible;
+
+        public System.Windows.Visibility IsPlaceholderVisible { get => isPlaceholderVisible; set => SetProperty(ref isPlaceholderVisible, value); }
```

Comment:
Zou je dit met niet window gerelateerde dingen kunnen oplossen? Ik zie dat je iets met visibility doet. Dit is beter om te doen met een bool in combinatie met een van de beschikbare converters.